### ISSUE TITLE
[In progress] DEEP LEARNING APPROACHES FOR ATTRIBUTE MANIPULATION AND TEXT-TO-IMAGE SYNTHESIS / Chapter4 : Attribute Manipulation Generative Adversarial Networks for Image-to-Image Translation

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#58]
## 0. 論文情報・リンク
- 論文リンク（４章のみの内容を要約）：https://www.researchgate.net/profile/Kenan_Ak/publication/340528919_DEEP_LEARNING_APPROACHES_FOR_ATTRIBUTE_MANIPULATION_AND_TEXT-TO-IMAGE_SYNTHESIS/links/5e8f05ab299bf1307989f746/DEEP-LEARNING-APPROACHES-FOR-ATTRIBUTE-MANIPULATION-AND-TEXT-TO-IMAGE-SYNTHESIS.pdf
- 公開日時：2019/11/25
- 被引用数（記事作成時点）：0 件
- 実装コード：なし
- Publication : 

## 1. どんなもの？
- 属性（服の色・袖の長さなど）を操作可能な２次元画像ベースの仮想試着モデル。CAMs [class activation maps] を利用して attention メカニズムを強化した AMGAN [Attribute Manipulation Generative Adversarial Networks]、及び識別器の属性関連領域限定の仕組みなどで実現している。
    <image src="https://user-images.githubusercontent.com/25688193/83634382-6b03fb80-a5dd-11ea-9f9b-048be03329d8.png" width="950">

## 2. 先行研究と比べてどこがすごいの？
- GANimation や SaGAN は Attention 構造のメカニズムを用いて、属性（顔の表情など）関連領域の変換をうまく行うことを実現している。しかしながらこれらの手法において attention mask を学習するための損失関数は、分類損失 [classification loss]（クロスエントロピーなど）で学習されるため、属性の数が増えるとうまく機能しなくなる。
	仮想試着タスクにおいて、GANimation のように属性操作可能なモデルを構築しようとすると、属性（服の色・袖の長さなど）の数が多いことが問題となる。
	本手法では、このような問題を解決するために CAMs [class activation maps] を利用して attention メカニズムを強化した AMGAN [Attribute Manipulation Generative Adversarial Networks]、及び識別器の属性関連領域限定の仕組みなどを導入している。
- 属性操作可能な試着モデルにすることで、ユーザーが属性（服の色・袖の長さなど）に満足できるように試着画像をカスタマイズすることができる。（この機能と画像検索機能を組み合わせることにより、カスタマイズした試着画像にマッチする試着画像を検索するといったことが可能となる）

## 3. 技術や手法の"キモ"はどこにある？
- AMGAN [Attribute Manipulation Generative Adversarial Networks] のアーキテクチャ
    <image src="https://user-images.githubusercontent.com/25688193/83634470-90910500-a5dd-11ea-9952-fc1d66157b09.png" width="780">
	上図は、AMGAN のネットワークのアーキテクチャ図である。
	この AMGAN ネットワークは、入力画像（人物画像）<image src="https://user-images.githubusercontent.com/25688193/83634646-d2ba4680-a5dd-11ea-9baa-41095b1c94fa.png" width="19"> と属性操作 m を入力として、その属性操作 m を反映した出力画像（仮想試着）<image src="https://user-images.githubusercontent.com/25688193/83634691-e6fe4380-a5dd-11ea-9bd0-b843b9d761b1.png" width="16"> を生成するネットワークになっている。
	ここで属性操作 m は、one-hot encode された以下の式で表現される。
        <image src="https://user-images.githubusercontent.com/25688193/83634604-cafaa200-a5dd-11ea-87d1-de06f37e53fd.png" width="500">
	この生成器は、学習イテレーション度に特定の属性に焦点を当てながら学習が行われる？

    - アテンションマスクとの合成        
		生成器ネットワークでは、入力画像（人物画像）<image src="https://user-images.githubusercontent.com/25688193/83634646-d2ba4680-a5dd-11ea-9baa-41095b1c94fa.png" width="19"> と属性操作 m を入力として、生成画像 z とアテンションマスクαを出力する。
		※ このアテンションマスクは GANimation のように特定の領域のみ属性操作の対象とし、他の領域は変更しなくするためのマスク
最終的な出力は、これら以下の式で２つを組み合わせたものとなる。        
        <image src="https://user-images.githubusercontent.com/25688193/83634777-0f863d80-a5de-11ea-839c-2d1eebc046d7.png" width="370">
    
    - CAMs でのマップ画像を利用したアテンションマスクの学習
		属性操作の目的は、属性を持つ領域を見つけ出し、それを新しい属性に変換することである。そのため生成器では、操作する属性に基づいて正しく領域を特定することが重要である。CNN から抽出した CAMs は、属性の識別領域を正しく定位させるために用いることができる。
		CAM でのマップ画像を正解画像として attention loss で学習することで、AMGAN の生成器はアテンションマスクを正しく生成することができ、結果的に属性操作能力を向上させることができる。
		具体的には、事前学習済み CNN に入力画像 <image src="https://user-images.githubusercontent.com/25688193/83634646-d2ba4680-a5dd-11ea-9baa-41095b1c94fa.png" width="19"> を入力することで、CAMs [class activation maps] に基づいたネットワーク活性化部分のマップ画像（ネットワークがどの部分を注視して学習しているのかを表すマップ画像）、及びガイダンスマスク画像 α∗ を生成することができる。
		そして、このガイダンスマスク画像 α∗  をアテンションマスク α の正解画像として利用することで、生成器にどの部分をアテンションすべきかを学習させることができる。（※ 詳細は生成器の損失関数の項目参照）
		※ SaGAN や GANimation でもアテンションマスクを生成していたが、CAMs からのマップ画像を正解画像として利用していなかった。この点が従来の手法とは異なる。

    - 属性の類似度に基づいた正解画像の割り当てと perceptual loss
		属性操作を行った試着画像に対する正解画像がないために、L1 loss や perceptual loss を定義できない。本手法ではこの問題を解決するために、属性の類似度に基づいて正解画像を割り当てることを行う。（※ 詳細は生成器の損失関数の項目参照）

    - 属性関連領域のみでの識別器
		一般的に識別器のネットワークは、画像全体に適用される対して本物 or 偽物を判定するネットワークになっている。
		本手法での識別器は、画像全体に適用されるネットワーク D_I と属性操作された領域のみに適用されるネットワーク D_C  で構成されている。
		属性操作された領域のみに適用されるネットワーク D_C  の入力 x_C^∗ は、入力画像 <image src="https://user-images.githubusercontent.com/25688193/83634646-d2ba4680-a5dd-11ea-9baa-41095b1c94fa.png" width="19"> と出力画像 <image src="https://user-images.githubusercontent.com/25688193/83634691-e6fe4380-a5dd-11ea-9bd0-b843b9d761b1.png" width="16"> に対してアテンションマスクαで crop させたものから算出された画像となる。（この crop 画像を入力することにより、属性操作された領域のみに適用されるネットワークになる）
		具体的には、アテンションマスクαの値が大きい（50%以上）画像を分割して、それらの領域を覆うバウンディングボックスを用いて、入力画像 or 出力画像を crop して、属性操作領域の小画像を切り出す。

    - 識別器の損失関数
        識別器の損失関数は、以下の式のように Adv loss と classification loss との線形結合で定義される。
        <image src="https://user-images.githubusercontent.com/25688193/83635039-802d5a00-a5de-11ea-88de-f73a800b8678.png" width="350">
        - GAN Adv loss
            <image src="https://user-images.githubusercontent.com/25688193/83635085-93402a00-a5de-11ea-9fb8-aab90d09417e.png" width="490">

        - classification loss
            <image src="https://user-images.githubusercontent.com/25688193/83635152-b074f880-a5de-11ea-97cd-77f0e421c078.png" width="380">
			本手法での識別器は real / fake の出力結果 <image src="https://user-images.githubusercontent.com/25688193/83635195-c4205f00-a5de-11ea-88f5-2a23bed32c13.png" width="133"> で本物画像と偽物画像を認証する以外にも、本物画像と認証画像における属性操作の分類を正しく分類出来るように classification 結果 <image src="https://user-images.githubusercontent.com/25688193/83635249-dbf7e300-a5de-11ea-81c1-3c2b2ade33f8.png" width="82"> も出力する。
			この本物画像に対する classification 結果を上式の cross entropy loss で学習することで、本物画像と認証画像における属性操作を正しく分類出来るようになる。
        
        - 生成器の損失関数
		生成器の損失関数は、以下の式のように adv loss,  classification loss, cycle consistency loss, attention loss, perceptual loss の線形結合で定義される
        <image src="https://user-images.githubusercontent.com/25688193/83635331-034eb000-a5df-11ea-944d-183ce2736ea2.png" width="450">

        - GAN Adv loss
            <image src="https://user-images.githubusercontent.com/25688193/83635387-1bbeca80-a5df-11ea-8b49-9559bb3a7576.png" width="360">

        - classification loss
            <image src="https://user-images.githubusercontent.com/25688193/83635455-385b0280-a5df-11ea-9c16-dd095390a1d6.png" width="340">
		属性操作 m を反映した画像を生成するために、識別器からの偽物画像に対する classification 結果 <image src="https://user-images.githubusercontent.com/25688193/83635490-4741b500-a5df-11ea-9303-be5a87fa508c.png" width="77"> に対して、上式の cross entropy loss で学習する

        - cycle consistency loss            
            <image src="https://user-images.githubusercontent.com/25688193/83635583-6f311880-a5df-11ea-92f0-6a55cc09c63c.png" width="360">
			属性操作の非対象領域に関しては出力画像が変更されないようにするため（人物同一性の確保）に、cycle consistency loss を導入する。具体的には、属性操作 m （服の色を白→赤など）とその反対属性操作 m∗（服の色を赤→白など）を連続で行った場合に、生成画像 <image src="https://user-images.githubusercontent.com/25688193/83635607-7e17cb00-a5df-11ea-9de7-91b1b0f377cd.png" width="100"> が元の画像 x_I  と等しくなるように働きかける。

            - attention loss
                <image src="https://user-images.githubusercontent.com/25688193/83635846-df3f9e80-a5df-11ea-840d-2af603188d23.png" width="440">
			属性の数が少ない場合は、上記の Adv loss + classification loss + cycle consistency loss で品質の高いアテンションマスクを生成出来るようになる。
			しかしながら属性の数が多い場合や姿勢の変化が大きいときは、上記損失関数のみではアテンションマスクの品質が不安定になる。
			本手法では上式のように、CAMs でのマップ画像を正解画像として損失関数を定義することでこの問題を解決している。
			具体的には、まず入力画像 x_I  を事前学習済み CNN に入力し、中間特徴マップ f_k を得る。このとき CAMs でのマップ画像 M は、各階層 k での特徴マップ f_k と重み w_k の線形結合で定義される。その後、そのマップ画像 M を [0,1] で正規化したガイダンスマスク α∗ が計算される。そして、そのガイダンスマスク α∗  と生成アテンションマスク α で L1 loss をとる。

            - perceptual loss
                <image src="https://user-images.githubusercontent.com/25688193/83635952-0f873d00-a5e0-11ea-96d8-bc607ea8f965.png" width="350">
			属性操作を行った試着画像に対する正解画像がないために、perceptual loss を定義できない。
			本手法ではこの問題を解決するために、属性操作後の出力画像とすべての属性が一致するような画像 x_ref を抽出し、それを正解画像として、それら画像間で上式の perceptual loss を定義している。
			※ x_I  と x_ref  の姿勢が異なっていても perceptual loss の計算を行う特徴マップでは両者は近いものとなるため、この方法は有効になっている。

## 4. どうやって有効だと検証した？
- xxx
        <image src="" width="">

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- xxx

## 7. 参考文献
- xxx
